---
layout: post
title: ClustalW2 Command line Arguments
date: '2011-07-29T16:02:00.001-07:00'
author: Mark Evans
tags:
- clustalw2
modified_time: '2011-07-29T16:07:34.700-07:00'
blogger_id: tag:blogger.com,1999:blog-8042776731714569314.post-1820981522344556822
blogger_orig_url: http://codexpython.blogspot.com/2011/07/clustalw2-command-line-arguments.html
---

CLUSTAL 2.0.12 Multiple Sequence Alignments<br /><br /><br />&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; DATA (sequences)<br /><br />-INFILE=file.ext &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; :input sequences.<br />-PROFILE1=file.ext &nbsp;and &nbsp;-PROFILE2=file.ext &nbsp;:profiles (old alignment).<br /><br /><br />&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; VERBS (do things)<br /><br />-OPTIONS &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;:list the command line parameters<br />-HELP &nbsp;or -CHECK &nbsp; &nbsp;:outline the command line params.<br />-FULLHELP &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; :output full help content.<br />-ALIGN &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;:do full multiple alignment.<br />-TREE &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; :calculate NJ tree.<br />-PIM &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;:output percent identity matrix (while calculating the tree)<br />-BOOTSTRAP(=n) &nbsp; &nbsp; &nbsp;:bootstrap a NJ tree (n= number of bootstraps; def. = 1000).<br />-CONVERT &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;:output the input sequences in a different file format.<br /><br /><br />&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; PARAMETERS (set things)<br /><br />***General settings:****<br />-INTERACTIVE :read command line, then enter normal interactive menus<br />-QUICKTREE &nbsp; :use FAST algorithm for the alignment guide tree<br />-TYPE= &nbsp; &nbsp; &nbsp; :PROTEIN or DNA sequences<br />-NEGATIVE &nbsp; &nbsp;:protein alignment with negative values in matrix<br />-OUTFILE= &nbsp; &nbsp;:sequence alignment file name<br />-OUTPUT= &nbsp; &nbsp; :GCG, GDE, PHYLIP, PIR or NEXUS<br />-OUTORDER= &nbsp; :INPUT or ALIGNED<br />-CASE &nbsp; &nbsp; &nbsp; &nbsp;:LOWER or UPPER (for GDE output only)<br />-SEQNOS= &nbsp; &nbsp; :OFF or ON (for Clustal output only)<br />-SEQNO_RANGE=:OFF or ON (NEW: for all output formats)<br />-RANGE=m,n &nbsp; :sequence range to write starting m to m+n<br />-MAXSEQLEN=n :maximum allowed input sequence length<br />-QUIET &nbsp; &nbsp; &nbsp; :Reduce console output to minimum<br />-STATS= &nbsp; &nbsp; &nbsp;:Log some alignents statistics to file<br /><br />***Fast Pairwise Alignments:***<br />-KTUPLE=n &nbsp; &nbsp;:word size<br />-TOPDIAGS=n &nbsp;:number of best diags.<br />-WINDOW=n &nbsp; &nbsp;:window around best diags.<br />-PAIRGAP=n &nbsp; :gap penalty<br />-SCORE &nbsp; &nbsp; &nbsp; :PERCENT or ABSOLUTE<br /><br /><br />***Slow Pairwise Alignments:***<br />-PWMATRIX= &nbsp; &nbsp;:Protein weight matrix=BLOSUM, PAM, GONNET, ID or filename<br />-PWDNAMATRIX= :DNA weight matrix=IUB, CLUSTALW or filename<br />-PWGAPOPEN=f &nbsp;:gap opening penalty &nbsp; &nbsp; &nbsp; <br />-PWGAPEXT=f &nbsp; :gap opening penalty<br /><br /><br />***Multiple Alignments:***<br />-NEWTREE= &nbsp; &nbsp; &nbsp;:file for new guide tree<br />-USETREE= &nbsp; &nbsp; &nbsp;:file for old guide tree<br />-MATRIX= &nbsp; &nbsp; &nbsp; :Protein weight matrix=BLOSUM, PAM, GONNET, ID or filename<br />-DNAMATRIX= &nbsp; &nbsp;:DNA weight matrix=IUB, CLUSTALW or filename<br />-GAPOPEN=f &nbsp; &nbsp; :gap opening penalty &nbsp; &nbsp; &nbsp; <br />-GAPEXT=f &nbsp; &nbsp; &nbsp;:gap extension penalty<br />-ENDGAPS &nbsp; &nbsp; &nbsp; :no end gap separation pen.<br />-GAPDIST=n &nbsp; &nbsp; :gap separation pen. range<br />-NOPGAP &nbsp; &nbsp; &nbsp; &nbsp;:residue-specific gaps off <br />-NOHGAP &nbsp; &nbsp; &nbsp; &nbsp;:hydrophilic gaps off<br />-HGAPRESIDUES= :list hydrophilic res. &nbsp; <br />-MAXDIV=n &nbsp; &nbsp; &nbsp;:% ident. for delay<br />-TYPE= &nbsp; &nbsp; &nbsp; &nbsp; :PROTEIN or DNA<br />-TRANSWEIGHT=f :transitions weighting<br />-ITERATION= &nbsp; &nbsp;:NONE or TREE or ALIGNMENT<br />-NUMITER=n &nbsp; &nbsp; :maximum number of iterations to perform<br />-NOWEIGHTS &nbsp; &nbsp; :disable sequence weighting<br /><br /><br />***Profile Alignments:***<br />-PROFILE &nbsp; &nbsp; &nbsp;:Merge two alignments by profile alignment<br />-NEWTREE1= &nbsp; &nbsp;:file for new guide tree for profile1<br />-NEWTREE2= &nbsp; &nbsp;:file for new guide tree for profile2<br />-USETREE1= &nbsp; &nbsp;:file for old guide tree for profile1<br />-USETREE2= &nbsp; &nbsp;:file for old guide tree for profile2<br /><br /><br />***Sequence to Profile Alignments:***<br />-SEQUENCES &nbsp; :Sequentially add profile2 sequences to profile1 alignment<br />-NEWTREE= &nbsp; &nbsp;:file for new guide tree<br />-USETREE= &nbsp; &nbsp;:file for old guide tree<br /><br /><br />***Structure Alignments:***<br />-NOSECSTR1 &nbsp; &nbsp; :do not use secondary structure-gap penalty mask for profile 1<br />-NOSECSTR2 &nbsp; &nbsp; :do not use secondary structure-gap penalty mask for profile 2<br />-SECSTROUT=STRUCTURE or MASK or BOTH or NONE &nbsp; :output in alignment file<br />-HELIXGAP=n &nbsp; &nbsp;:gap penalty for helix core residues<br />-STRANDGAP=n &nbsp; :gap penalty for strand core residues<br />-LOOPGAP=n &nbsp; &nbsp; :gap penalty for loop regions<br />-TERMINALGAP=n :gap penalty for structure termini<br />-HELIXENDIN=n &nbsp;:number of residues inside helix to be treated as terminal<br />-HELIXENDOUT=n :number of residues outside helix to be treated as terminal<br />-STRANDENDIN=n :number of residues inside strand to be treated as terminal<br />-STRANDENDOUT=n:number of residues outside strand to be treated as terminal<br /><br /><br />***Trees:***<br />-OUTPUTTREE=nj OR phylip OR dist OR nexus<br />-SEED=n &nbsp; &nbsp; &nbsp; &nbsp;:seed number for bootstraps.<br />-KIMURA &nbsp; &nbsp; &nbsp; &nbsp;:use Kimura's correction. <br />-TOSSGAPS &nbsp; &nbsp; &nbsp;:ignore positions with gaps.<br />-BOOTLABELS=node OR branch :position of bootstrap values in tree display<br />-CLUSTERING= &nbsp; :NJ or UPGMA<br /><div><br /><span class="Apple-style-span" style="font-family: 'Segoe UI', Calibri, 'Myriad Pro', Myriad, 'Trebuchet MS', Helvetica, Arial, sans-serif; font-size: 15px; line-height: 22px;"></span><br /><div style="border-bottom-width: 0px; border-color: initial; border-left-width: 0px; border-right-width: 0px; border-style: initial; border-top-width: 0px; font-family: inherit; font-size: 15px; font-style: inherit; font-weight: inherit; margin-bottom: 20px; margin-left: 0px; margin-right: 0px; margin-top: 0px; padding-bottom: 0px; padding-left: 0px; padding-right: 0px; padding-top: 0px; vertical-align: baseline;">&gt;&gt; HELP 0 &lt;&lt;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Help for tree output format options</div><div style="border-bottom-width: 0px; border-color: initial; border-left-width: 0px; border-right-width: 0px; border-style: initial; border-top-width: 0px; font-family: inherit; font-size: 15px; font-style: inherit; font-weight: inherit; margin-bottom: 20px; margin-left: 0px; margin-right: 0px; margin-top: 0px; padding-bottom: 0px; padding-left: 0px; padding-right: 0px; padding-top: 0px; vertical-align: baseline;">Four output formats are offered: 1) Clustal, 2) Phylip, 3) Just the distances<br />4) Nexus</div><div style="border-bottom-width: 0px; border-color: initial; border-left-width: 0px; border-right-width: 0px; border-style: initial; border-top-width: 0px; font-family: inherit; font-size: 15px; font-style: inherit; font-weight: inherit; margin-bottom: 20px; margin-left: 0px; margin-right: 0px; margin-top: 0px; padding-bottom: 0px; padding-left: 0px; padding-right: 0px; padding-top: 0px; vertical-align: baseline;">None of these formats displays the results graphically. Many packages can<br />display trees in the the PHYLIP format 2) below. It can also be imported into<br />the PHYLIP programs RETREE, DRAWTREE and DRAWGRAM for graphical display.<br />NEXUS format trees can be read by PAUP and MacClade.</div><div style="border-bottom-width: 0px; border-color: initial; border-left-width: 0px; border-right-width: 0px; border-style: initial; border-top-width: 0px; font-family: inherit; font-size: 15px; font-style: inherit; font-weight: inherit; margin-bottom: 20px; margin-left: 0px; margin-right: 0px; margin-top: 0px; padding-bottom: 0px; padding-left: 0px; padding-right: 0px; padding-top: 0px; vertical-align: baseline;">1) Clustal format output.<br />This format is verbose and lists all of the distances between the sequences and<br />the number of alignment positions used for each. The tree is described at the<br />end of the file. It lists the sequences that are joined at each alignment step<br />and the branch lengths. After two sequences are joined, it is referred to later<br />as a NODE. The number of a NODE is the number of the lowest sequence in that<br />NODE.</div><div style="border-bottom-width: 0px; border-color: initial; border-left-width: 0px; border-right-width: 0px; border-style: initial; border-top-width: 0px; font-family: inherit; font-size: 15px; font-style: inherit; font-weight: inherit; margin-bottom: 20px; margin-left: 0px; margin-right: 0px; margin-top: 0px; padding-bottom: 0px; padding-left: 0px; padding-right: 0px; padding-top: 0px; vertical-align: baseline;">2) Phylip format output.<br />This format is the New Hampshire format, used by many phylogenetic analysis<br />packages. It consists of a series of nested parentheses, describing the<br />branching order, with the sequence names and branch lengths. It can be used by<br />the RETREE, DRAWGRAM and DRAWTREE programs of the PHYLIP package to see the<br />trees graphically. This is the same format used during multiple alignment for<br />the guide trees.</div><div style="border-bottom-width: 0px; border-color: initial; border-left-width: 0px; border-right-width: 0px; border-style: initial; border-top-width: 0px; font-family: inherit; font-size: 15px; font-style: inherit; font-weight: inherit; margin-bottom: 20px; margin-left: 0px; margin-right: 0px; margin-top: 0px; padding-bottom: 0px; padding-left: 0px; padding-right: 0px; padding-top: 0px; vertical-align: baseline;">Use this format with NJplot (Manolo Gouy), supplied with Clustal W. Some other<br />packages that can read and display New Hampshire format are TreeView (Mac/PC),<br />TreeTool (UNIX), and Phylowin.</div><div style="border-bottom-width: 0px; border-color: initial; border-left-width: 0px; border-right-width: 0px; border-style: initial; border-top-width: 0px; font-family: inherit; font-size: 15px; font-style: inherit; font-weight: inherit; margin-bottom: 20px; margin-left: 0px; margin-right: 0px; margin-top: 0px; padding-bottom: 0px; padding-left: 0px; padding-right: 0px; padding-top: 0px; vertical-align: baseline;">3) The distances only.<br />This format just outputs a matrix of all the pairwise distances in a format<br />that can be used by the Phylip package. It used to be useful when one could not<br />produce distances from protein sequences in the Phylip package but is now<br />redundant (Protdist of Phylip 3.5 now does this).</div><div style="border-bottom-width: 0px; border-color: initial; border-left-width: 0px; border-right-width: 0px; border-style: initial; border-top-width: 0px; font-family: inherit; font-size: 15px; font-style: inherit; font-weight: inherit; margin-bottom: 20px; margin-left: 0px; margin-right: 0px; margin-top: 0px; padding-bottom: 0px; padding-left: 0px; padding-right: 0px; padding-top: 0px; vertical-align: baseline;">4) NEXUS FORMAT TREE. This format is used by several popular phylogeny programs,<br />including PAUP and MacClade. The format is described fully in:<br />Maddison, D. R., D. L. Swofford and W. P. Maddison.&nbsp; 1997.<br />NEXUS: an extensible file format for systematic information.<br />Systematic Biology 46:590-621.</div><div style="border-bottom-width: 0px; border-color: initial; border-left-width: 0px; border-right-width: 0px; border-style: initial; border-top-width: 0px; font-family: inherit; font-size: 15px; font-style: inherit; font-weight: inherit; margin-bottom: 20px; margin-left: 0px; margin-right: 0px; margin-top: 0px; padding-bottom: 0px; padding-left: 0px; padding-right: 0px; padding-top: 0px; vertical-align: baseline;">5) TOGGLE PHYLIP BOOTSTRAP POSITIONS<br />By default, the bootstrap values are placed on the nodes of the phylip format<br />output tree. This is inaccurate as the bootstrap values should be associated<br />with the tree branches and not the nodes. However, this format can be read and<br />displayed by TreeTool, TreeView and Phylowin. An option is available to<br />correctly place the bootstrap values on the branches with which they are<br />associated.</div></div>