---
layout: post
title: BioPython PSSM Matrices
date: '2010-12-02T14:18:00.001-08:00'
author: Mark Evans
tags: 
modified_time: '2010-12-02T15:47:47.360-08:00'
blogger_id: tag:blogger.com,1999:blog-8042776731714569314.post-7128112505932579392
---

I have been confused about the difference between the PSSM that is generated from a sequence alignment using Bio.Align.AlignInfo.SummaryInfo.pos_specific_scoring_matrix() and the Bio.Motif package. &nbsp;I have always used the PSSM from Bio.Align but it is a simple occurrence matrix and it doesn't seem to have any built in ability to convert to a frequency or log-likelihood matrix.<br /><br />The Bio.Motif matrix does have these abilities, but all of the examples in the BioPython docs and all I could find on the web only talk about using it for DNA motifs, searching the TRANSFAC database, etc. which made me think it would not work for what I want to do. &nbsp;Plus I was not used to thinking about PSSM's in terms of "motifs", as I have always used them from "alignments". &nbsp;However, it appears that it does indeed work with proteins, you just kind of have to figure it out on your own. &nbsp;So that is what this is about.<br /><br />To create a motif object for proteins, we will do the following:<br /><br /><span class="Apple-style-span" style="color: #404040; font-family: Arial, Tahoma, Helvetica, FreeSans, sans-serif; font-size: 13px; line-height: 18px;"></span><br /><pre class="prettyprint" style="background-attachment: initial; background-clip: initial; background-color: #eeeeee; background-image: initial; background-origin: initial; border-bottom-width: 0px; border-color: initial; border-left-width: 0px; border-right-width: 0px; border-style: initial; border-top-width: 0px; font-family: Consolas, Menlo, Monaco, 'Lucida Console', 'Liberation Mono', 'DejaVu Sans Mono', 'Bitstream Vera Sans Mono', 'Courier New', monospace, serif; font-size: 14px; margin-bottom: 10px; margin-left: 0px; margin-right: 0px; margin-top: 0px; max-height: 600px; overflow-x: auto; overflow-y: auto; padding-bottom: 5px; padding-left: 5px; padding-right: 5px; padding-top: 5px; vertical-align: baseline; width: auto;"><span class="Apple-style-span" style="color: black;">from Bio import Motif                                                              from Bio.Alphabet import IUPAC                     &nbsp;                               m = Motif.Motif(alphabet=IUPAC.protein)</span></pre>Note that I am not using the extended_protein alphabet. &nbsp;If you do this, then non-standard amino acids will be contained in the matrix. &nbsp;Currently the matrix object is empty so we need to fill it. &nbsp;Normally I would load an alignment object if I was doing this through the SummaryInfo method. &nbsp;However, in this case, individual sequences need to be loaded. &nbsp;So I am going to use the multi-sequence FASTA file as a source, rather than the corresponding alignment itself<br /><br /><span class="Apple-style-span" style="color: #404040; font-family: Arial, Tahoma, Helvetica, FreeSans, sans-serif; font-size: 13px; line-height: 18px;"></span><br /><pre class="prettyprint" style="background-attachment: initial; background-clip: initial; background-color: #eeeeee; background-image: initial; background-origin: initial; border-bottom-width: 0px; border-color: initial; border-left-width: 0px; border-right-width: 0px; border-style: initial; border-top-width: 0px; font-family: Consolas, Menlo, Monaco, 'Lucida Console', 'Liberation Mono', 'DejaVu Sans Mono', 'Bitstream Vera Sans Mono', 'Courier New', monospace, serif; font-size: 14px; margin-bottom: 10px; margin-left: 0px; margin-right: 0px; margin-top: 0px; max-height: 600px; overflow-x: auto; overflow-y: auto; padding-bottom: 5px; padding-left: 5px; padding-right: 5px; padding-top: 5px; vertical-align: baseline; width: auto;"><code style="background-attachment: initial; background-clip: initial; background-color: #eeeeee; background-image: initial; background-origin: initial; border-bottom-width: 0px; border-color: initial; border-left-width: 0px; border-right-width: 0px; border-style: initial; border-top-width: 0px; font-family: Consolas, Menlo, Monaco, 'Lucida Console', 'Liberation Mono', 'DejaVu Sans Mono', 'Bitstream Vera Sans Mono', 'Courier New', monospace, serif; font-size: 14px; margin-bottom: 0px; margin-left: 0px; margin-right: 0px; margin-top: 0px; padding-bottom: 0px; padding-left: 0px; padding-right: 0px; padding-top: 0px; vertical-align: baseline;"><span class="pln" style="background-attachment: initial; background-clip: initial; background-color: transparent; background-image: initial; background-origin: initial; border-bottom-width: 0px; border-color: initial; border-left-width: 0px; border-right-width: 0px; border-style: initial; border-top-width: 0px; color: black; font-size: 14px; margin-bottom: 0px; margin-left: 0px; margin-right: 0px; margin-top: 0px; padding-bottom: 0px; padding-left: 0px; padding-right: 0px; padding-top: 0px; vertical-align: baseline;">from Bio import SeqIO                                                              for record in SeqIO.parse("myseqs.fasta","fasta"):                                    m.add_instance(record.seq,m.alphabet)                                    </span></code></pre>